src/Compilers/VisualBasic/Portable/Utilities/FeatureUtils.vb

```diff
@@ -0,0 +1,79 @@
+        <Runtime.CompilerServices.Extension>
+        Private Function IsInLanguageVersion(feature As Feature, options As VisualBasicParseOptions) As Boolean
+            Dim required = feature.GetLanguageVersion()
+            Dim current = options.LanguageVersion
+            Return required <= current
+        End Function
+
+        ''' <summary>Report the unavailibility of a language feature.</summary>
+        ''' <typeparam name="TNode"></typeparam>
+        ''' <param name="feature">Language feature to report as unavailable.</param>
+        ''' <param name="parseOptions">The parse options being used.</param>
+        ''' <param name="node">The node to attach the diagnostic (Feature Unavailable).</param>
+        ''' <returns>Return the node with the diagnostic attached to it.</returns>
+        <Runtime.CompilerServices.Extension>
+        Public Function ReportFeatureUnavailable(Of TNode As VisualBasicSyntaxNode)(feature As Feature, parseOptions As VisualBasicParseOptions, node As TNode) As TNode
+            Dim featureName = ErrorFactory.ErrorInfo(feature.GetResourceId())
+            Dim requiredVersion As New VisualBasicRequiredLanguageVersion(feature.GetLanguageVersion())
+            Return Parser.ReportSyntaxError(node, ERRID.ERR_LanguageVersion, parseOptions.LanguageVersion.GetErrorName(), featureName, requiredVersion)
+        End Function
+
+        ''' <summary>
+        ''' Check to see if the given <paramref name="feature"/> is available with the <see cref="LanguageVersion"/>
+        ''' of the parser.
+        ''' </summary>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="parseOptions">The parse options being used.</param>
+        ''' <returns></returns>
+        <Runtime.CompilerServices.Extension>
+        Public Function IsAvailable(feature As Feature, parseOptions As VisualBasicParseOptions) As Boolean
+            Dim featureFlag = feature.GetFeatureFlag()
+            Return ((featureFlag IsNot Nothing) AndAlso parseOptions.Features.ContainsKey(featureFlag)) OrElse feature.IsInLanguageVersion(parseOptions)
+        End Function
+
+        ''' <summary>
+        '''  Check to see if the given <paramref name="feature"/> is available with the <see cref="LanguageVersion"/>
+        ''' of the parser.  If it is not available a diagnostic will be added to the returned value.
+        ''' </summary>
+        ''' <param name="feature">Language feature to check is available.</param>
+        ''' <param name="parseoptions">The parse options being used.</param>
+        ''' <param name="node">The node to attach the potential diagnostic (Feature Unavailable).</param>
+        ''' <returns></returns>
+        <Runtime.CompilerServices.Extension>
+        Public Function IsAvailable(Of TNode As VisualBasicSyntaxNode)(feature As Feature, parseoptions As VisualBasicParseOptions, node As TNode) As TNode
```

Comment:
The `VisualBasicParseOptions` argument is named differently in these methods: `options`, `parseOptions`, `parseoptions`. Perhaps just `options` everywhere.